packages/smart-vaults/src/utils/LightSyncMultiSigner.sol

```diff
@@ -142,84 +112,57 @@ abstract contract LightSyncMultiSigner is MultiSigner {
         }
     }
 
     function _processSignerSetUpdateMemory(
-        SignerSetUpdate memory signerUpdate_,
-        bytes memory signers_,
-        uint8 threshold_
+        uint256 insertIndex_,
+        SignerSetUpdate memory signerSetUpdate_,
+        uint256 memorySignerSet_,
+        bytes memory signerUpdates_
     )
         internal
         view
-        returns (bytes memory, uint8)
+        returns (uint256)
     {
-        SignerUpdateParam[] memory updateParams = signerUpdate_.updateParams;
-        uint256 numUpdates = updateParams.length;
+        (bytes memory signer, uint8 index) = abi.decode(signerSetUpdate_.data, (bytes, uint8));
+        MultiSignerLib.validateSigner(signer);
 
-        SignerUpdateParam memory signerUpdateParam;
-        for (uint256 i; i < numUpdates; i++) {
-            signerUpdateParam = updateParams[i];
-            if (signerUpdateParam.updateType == SignerUpdateType.AddSigner) {
-                (bytes memory signer, uint8 index) = abi.decode(signerUpdateParam.data, (bytes, uint8));
-                MultiSignerLib.validateSigner(signer);
-
-                uint8 signerType;
-                if (signer.length == MultiSignerLib.EOA_SIGNER_SIZE) {
-                    signerType = MultiSignerSignatureLib.EOA_SIGNER_TYPE;
-                } else {
-                    signerType = MultiSignerSignatureLib.PASSKEY_SIGNER_TYPE;
-                }
-
-                uint256 start = uint256(index) * MultiSignerSignatureLib.SIGNER_SIZE;
-                assembly {
-                    let dataPtr := add(signers_, add(32, start))
-                    mstore8(dataPtr, signerType)
-
-                    let newDataPtr := add(signer, 32)
-                    mstore(add(dataPtr, 1), mload(newDataPtr))
-
-                    if eq(signerType, 2) { mstore(add(dataPtr, 33), mload(add(newDataPtr, 32))) }
-                }
-            } else if (signerUpdateParam.updateType == SignerUpdateType.RemoveSigner) {
-                uint8 index = abi.decode(signerUpdateParam.data, (uint8));
-                uint8 signerType = MultiSignerSignatureLib.REMOVED_SIGNER_TYPE;
-
-                uint256 start = uint256(index) * MultiSignerSignatureLib.SIGNER_SIZE;
-                assembly {
-                    let dataPtr := add(signers_, add(32, start))
-                    mstore8(dataPtr, signerType)
-                }
-            } else if (signerUpdateParam.updateType == SignerUpdateType.UpdateThreshold) {
-                threshold_ = abi.decode(signerUpdateParam.data, (uint8));
-            } else {
-                revert InvalidSignerUpdateParam(signerUpdateParam);
-            }
+        uint256 bitMask = 1 << index;
+        if (memorySignerSet_ & bitMask != 0) {
+            revert SignerAlreadyPresent(index);
         }
+        memorySignerSet_ = memorySignerSet_ | bitMask;
 
-        return (signers_, threshold_);
-    }
+        uint8 signerType;
```

Comment:
i think this can be cleaned up a bit from here down but don't want to hold up submitting review